driver-async/src/main/com/mongodb/async/client/MongoClientSettings.java

```diff
@@ -80,6 +82,7 @@ public static Builder builder(final MongoClientSettings settings) {
     public static final class Builder {
         private List<MongoCredential> credentialList = Collections.emptyList();
         private final com.mongodb.MongoClientSettings.Builder wrappedBuilder = com.mongodb.MongoClientSettings.builder();
+        private final SocketSettings.Builder heartbeatSocketSettingsBuilder = SocketSettings.builder().readTimeout(20000, MILLISECONDS);
```

Comment:
Both connectTimeout and readTimeout should default to the eventual connectTimeout of wrapped.socketSettings, unless overridden by heartbeatSocketSettings below.